src/engine/sim_engine/engine.rs

```diff
@@ -221,41 +221,44 @@ impl Engine for SimEngine {
 
     async fn get_pool(
         &self,
-        key: LockKey<PoolUuid>,
+        key: PoolIdentifier<PoolUuid>,
     ) -> Option<SomeLockReadGuard<PoolUuid, Self::Pool>> {
         get_pool!(self; key)
     }
 
     async fn get_mut_pool(
         &self,
-        key: LockKey<PoolUuid>,
+        key: PoolIdentifier<PoolUuid>,
     ) -> Option<SomeLockWriteGuard<PoolUuid, Self::Pool>> {
         get_mut_pool!(self; key)
     }
 
-    async fn locked_pools(&self) -> HashMap<PoolUuid, LockedPoolInfo> {
-        HashMap::new()
-    }
-
-    async fn stopped_pools(&self) -> HashMap<PoolUuid, StoppedPoolInfo> {
-        let mut map = HashMap::new();
-        for (_, uuid, pool) in self.stopped_pools.read().await.iter() {
-            map.insert(
-                *uuid,
-                StoppedPoolInfo {
-                    info: pool.encryption_info(),
-                    devices: pool
-                        .blockdevs()
-                        .into_iter()
-                        .map(|(dev_uuid, _, bd)| PoolDevice {
-                            devnode: bd.devnode().to_path_buf(),
-                            uuid: dev_uuid,
-                        })
-                        .collect::<Vec<_>>(),
-                },
-            );
-        }
-        map
+    async fn locked_pools(&self) -> LockedPoolsInfo {
+        LockedPoolsInfo::default()
+    }
+
+    async fn stopped_pools(&self) -> StoppedPoolsInfo {
+        self.stopped_pools.read().await.iter().fold(
+            StoppedPoolsInfo::default(),
+            |mut st, (name, uuid, pool)| {
+                st.name_to_uuid.insert(name.clone(), *uuid);
+                st.stopped.insert(
+                    *uuid,
+                    StoppedPoolInfo {
+                        info: pool.encryption_info(),
+                        devices: pool
+                            .blockdevs()
+                            .into_iter()
+                            .map(|(dev_uuid, _, bd)| PoolDevice {
+                                devnode: bd.devnode().to_path_buf(),
+                                uuid: dev_uuid,
+                            })
+                            .collect::<Vec<_>>(),
```

Comment:
Please add updates for uuid_to_name in sim engine also.

Reply:
Okay, this is resolved!